sds_data_manager/stacks/ialirt_ingest_lambda_stack.py

```diff
@@ -0,0 +1,124 @@
+        construct_id: str,
+        env: cdk.Environment,
+        packet_data_table: aws_dynamodb.Table,
+        ialirt_bucket: aws_s3.Bucket,
+        **kwargs,
+    ) -> None:
+        """IalirtIngestLambda Stack.
+
+        Parameters
+        ----------
+        scope : Construct
+            Parent construct.
+        construct_id : str
+            A unique string identifier for this construct.
+        env : obj
+            Environment.
+        packet_data_table : aws_dynamodb.Table
+            Database table.
+        ialirt_bucket : aws_s3.Bucket
+            The data bucket.
+        kwargs : dict
+            Keyword arguments.
+
+        """
+        super().__init__(scope, construct_id, env=env, **kwargs)
+
+        lambda_role = iam.Role(
+            self,
+            "IalirtIngestLambdaRole",
+            assumed_by=iam.ServicePrincipal("lambda.amazonaws.com"),
+            managed_policies=[
+                iam.ManagedPolicy.from_aws_managed_policy_name(
+                    "service-role/AWSLambdaBasicExecutionRole"
+                )
+            ],
+        )
+
+        lambda_role.add_to_policy(
+            iam.PolicyStatement(
+                actions=[
+                    "dynamodb:PutItem",
+                    "dynamodb:BatchGetItem",
+                    "dynamodb:DescribeStream",
+                    "dynamodb:DescribeTable",
+                    "dynamodb:GetItem",
+                    "dynamodb:ListStreams",
+                    "dynamodb:ListTables",
+                    "dynamodb:Query",
+                    "dynamodb:Scan",
+                    "dynamodb:UpdateItem",
+                    "dynamodb:DeleteItem",
+                    "s3:GetObject",
+                ],
+                resources=[
+                    packet_data_table.table_arn,
+                    f"{ialirt_bucket.bucket_arn}/*",
+                ],
+            )
+        )
+
+        ialirt_ingest_lambda = lambda_alpha_.PythonFunction(
+            self,
+            id="IalirtIngestLambda",
+            function_name="ialirt-ingest",
+            entry=str(
+                pathlib.Path(__file__).parent.joinpath("..", "lambda_code").resolve()
+            ),
+            index="IAlirtCode/ialirt_ingest.py",
+            handler="lambda_handler",
+            runtime=lambda_.Runtime.PYTHON_3_9,
+            timeout=cdk.Duration.minutes(1),
+            memory_size=1000,
+            role=lambda_role,
+            environment={
+                "TABLE_NAME": packet_data_table.table_name,
+                "S3_BUCKET": ialirt_bucket.bucket_name,
+            },
+        )
+
+        packet_data_table.grant_read_write_data(ialirt_ingest_lambda)
+
+        # The resource is deleted when the stack is deleted.
+        ialirt_ingest_lambda.apply_removal_policy(cdk.RemovalPolicy.DESTROY)
+
+        # Event that triggers Lambda:
+        # Arrival of packets in s3 bucket.
+        ialirt_data_arrival_rule = events.Rule(
```

Comment:
I thought there was potentially also an option for you to send ops-software a python script that would send data to dynamodb directly that they could invoke from their tools as a subprocess. Is every packet going to be saved as an individual file in this scenario, or is there going to be a pool of packets in a single s3 file that you'll need to parse later?

Living within their container as `write_to_dynamodb.py`:
```python
import boto3
table = boto3.resource('dynamodb').Table(TABLE_NAME)
table.put_item(
       Item={
            'spacecraft_time': time,
            "packet": packet_data_from_command_line
        }
    )
```

Reply:
We talked about this a while back and Steve Monk and Evan Brookens said they wanted to mount to s3, which was what the previous pr was about. Each file in s3 contains multiple packets.

Reply:
Chatted about this on a call and I was mistaken about the mapping here between s3 and dynamodb.

s3 will have a 1-minute pkts file written to it containing multiple packets. This function will then loop over those packets and extract them into DynamoDB, which will have one entry per packet.

Ignore this suggestion.